src/components/Blocks/Flickr/DefaultView.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import SocialContentWrapper from '../../SocialContentWrapper/SocialContentWrapper';
+
+const FlickrView = (props) => {
+  const { flickrId, align } = props;
+  const linkText = 'View post on Flickr';
+  const parser = __CLIENT__ ? new DOMParser() : undefined;
+  const galleryData = parser?.parseFromString(flickrId, 'text/html');
+  const scriptSrc = '//embedr.flickr.com/assets/client-code.js';
+
+  React.useEffect(() => {
+    const head = document.querySelector('head');
+    const script = document.createElement('script');
+
+    script.setAttribute('src', scriptSrc);
+    head.appendChild(script);
+
+    return () => {
+      head.removeChild(script);
+    };
+  }, [scriptSrc]);
+
+  return (
+    flickrId && (
+      <SocialContentWrapper
+        align={align}
+        tool="flickr"
+        url={flickrId}
+        linkText={linkText}
+      >
+        <figure className="flickr-content">
+          <a
+            data-flickr-embed={true}
+            data-footer={galleryData?.links[0].dataset.footer}
+            data-header={galleryData?.links[0].dataset.header}
+            href={galleryData?.links[0].href ?? ''}
+            title={galleryData?.links[0].title}
+          >
+            <img
+              src={galleryData?.images[0].src}
+              width="100%"
+              alt={galleryData?.images[0].alt ?? ''}
+            />
```

Comment:
@danalvrz Try it with an embed code that has been modified to remove the a tag or remove the img. Then I think this will break when it tries to get `links[0]` or `images[0]`.

Reply:
`updateFlickrId` prevents the state from being updated if the value isn't valid, so an incomplete embed code wouldn't get to that point. If it's a new block the input box has a color alert when invalid, and if updating a block on the sidebar, it just wouldn't change the old valid value

Reply:
@danalvrz That makes sense.